mprs/src/parser.py

```diff
@@ -0,0 +1,175 @@
+        'name': name,
+        'role': role,
+        'type': type_e,
+        'workplace': workplace,
+        'active': active_e,
+        "income": {
+            'total': income_total,
+            'wage': wage,
+            'perks': {
+                'total': perks_total,
+            },
+            'other': { 
+                #Gratificações
+                'total': gratifications_total,
+                'trust_position': trust_position,
+                'others_total': others_total,
+                'others': {
+                    'Gratificação Natalina': others_christmas_bonus,
+                    'Férias (1/3 constitucional)': others_vacation,
+                    'Abono de Permanência': other_allowance_of_permanence,
+                }
+            },
+        },
+        'discounts': {
+            'total': discounts_total,
+            'prev_contribution': discounts_prev_contribution,
+            'ceil_retention': discounts_ceil_retention,
+            'income_tax': discounts_income_tax
+        }
+    }
+
+    return employee
+
+def update_employees(file_name, employees):
+    rows = read_data(file_name)
+
+    for row in rows["response"]["docs"]:
+        reg = row["numfunc_s"]
+        income_total = float(row["total_creditos_tf"])
+        wage = float(row["outras_verbas_tf"]) + float(row["vencimentos_tf"])
+        # Indemnities
+        perks_total = float(row["indenizacoes_tf"])
+        # Gratifications 
+        gratifications_total = float(row["fg_cc_tf"]) + float(row["gratificacao_natalina_tf"]) + float(row["ferias_tf"]) + float(row["abono_permanencia_tf"])
+        trust_position = float(row["fg_cc_tf"])
+        others_total = float(row["gratificacao_natalina_tf"]) + float(row["ferias_tf"]) + float(row["abono_permanencia_tf"])
+        others_christmas_bonus = float(row["gratificacao_natalina_tf"])
+        others_vacation = float(row["ferias_tf"])
+        other_allowance_of_permanence = float(row["abono_permanencia_tf"])
+        # Discounts
+        discounts_total = abs(float(row["total_descontos_tf"]))
+        discounts_prev_contribution = abs(float(row["previdencia_tf"]))
+        discounts_ceil_retention = abs(float(row["estorno_de_teto_tf"]))
+        discounts_income_tax = abs(float(row["ir_tf"]))
+        
+        try:
+            emp = employees[reg]
+        except Exception:
+            employee = generates_employee(row, file_name)
+            employees[reg] = employee
+            print(employee)
+            continue
+
+        emp['income'].update({
+            'total': emp['income']['total'] + income_total,
+            'wage': emp['income']['wage'] + wage
+        })
+        emp['income']['perks'].update({
+            'total': emp['income']['perks']['total'] + perks_total
+        })
+        emp['income']['other'].update({
+            'total': emp['income']['other']['total'] + gratifications_total,
+            'trust_position': emp['income']['other']['trust_position'] + trust_position,
```

Comment:
Você quer somar o que já tinha no campo com o que tem agora? Tem certeza que isso precisa ser assim? Se o que tinha antes era zero, não precisa essa primeira parcela do somatório.

Mesma coisa para os demais campos.

Poderia checar,por favor?

Reply:
É que eventualmente o valor que tinha antes não vai ser 0. Por exemplo: Um membro que recebe um bônus por cargo de confiança vai ter algum valor nesse campo quando eu criar o employee e o inserir no dicionário (a partir da folha normal). Quando eu for fazer a atualização com a folha de 13º salário, também vai haver o valor extra pelo cargo de confiança, daí vem a necessidade da soma. O mesmo acontece com os descontos, e com indenizações na folha complementar (e consequentemente com os valores "totais"). Não achei (buscando no site) a ocorrência com as gratificações, mas acredito que vale a pena deixar a soma também nesses campos, pra caso ela eventualmente seja necessária.

Reply:
Entendo. Pode colocar um comentário explicando isso, por favor?

Comentei pois algumas folhas normais já colocam o total de verbas indenizatórias na folha principal e só detalham na planilha específica de verbas indenizatórias. Se fosse esse o caso, você estaria somando duas vezes.